### PR TITLE
release container build fixes

### LINKED ISSUE #166: [FEA] Add CMake Option to Disable Pybind11 Stub Generation
**Is your feature request related to a problem? Please describe.**
Generating stubs during the build requires loading the NVIDIA driver. If that is not available during build (for example, during a docker build), then the build process will fail.

**Describe the solution you'd like**
Add a CMake option `MORPHEUS_BUILD_PYTHON_STUBS` that disables building the stubs when set to `OFF` but defaults to `ON`

**Describe alternatives you've considered**
This would not be necessary if the docker buildkit runtime could be changed to use nvidia during the build. Current issue is tracking here: https://github.com/moby/buildkit/issues/1436

### PR DESCRIPTION
**Pull request by cwharris**
Changes required to build the release container.

Closes #166

### CODE REVIEW

**Review by pdmack — COMMENTED** [review]
These will conflict with #152 so merge sequencing question. @mdemoret-nv

**Comment by pdmack** [comment]
CMAKE_CUDA_ARCHITECTURES var issue
```
#19 788.0 INFO:conda_build.build:Packaging morpheus-22.04.00-cuda_11.5_py38_g2d493e0_73
#19 788.0 Packaging morpheus-22.04.00-cuda_11.5_py38_g2d493e0_73
#19 788.7 Traceback (most recent call last):
#19 788.7   File "/opt/conda/bin/conda-mambabuild", line 10, in <module>
#19 788.7     sys.exit(main())
#19 788.7   File "/opt/conda/lib/python3.8/site-packages/boa/cli/mambabuild.py", line 239, in main
#19 788.7     call_conda_build(action, config)
#19 788.7   File "/opt/conda/lib/python3.8/site-packages/boa/cli/mambabuild.py", line 211, in call_conda_build
#19 788.7     result = api.build(
#19 788.7   File "/opt/conda/lib/python3.8/site-packages/conda_build/api.py", line 186, in build
#19 788.7     return build_tree(
#19 788.7   File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 3088, in build_tree
#19 788.7     packages_from_this = build(metadata, stats,
#19 788.7   File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 2371, in build
#19 788.7     newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
#19 788.7   File "/opt/conda/lib/python3.8/site-packages/conda_build/build.py", line 1602, in bundle_conda
#19 788.7     raise ValueError("env var '{}' specified in script_env, but is not set."
#19 788.7 ValueError: env var 'CMAKE_CUDA_ARCHITECTURES' specified in script_env, but is not set.
------
error: failed to solve: executor failed running [/bin/bash -c source activate base &&    MORPHEUS_ROOT=/workspace MORPHEUS_BUILD_PYTHON_STUBS=OFF CONDA_BLD_PATH=/opt/conda/conda-bld CONDA_ARGS="--no-test" ./ci/conda/recipes/run_conda_build.sh morpheus]: exit code: 1
```

File: ci/conda/recipes/morpheus/meta.yaml [hunk 1]
```diff
@@ -28,46 +28,16 @@ source:
...
   - name: morpheus
     build:
       number: {{ GIT_DESCRIBE_NUMBER }}
       string: cuda_{{ cuda_version }}_py{{ py_version }}_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
       script_env:
-        - CMAKE_CUDA_ARCHITECTURES=ALL
+        # Note: Setting a default value here will overwrite the value from the
+        # shell! Defaults should be set in the build script
+        - CMAKE_CUDA_ARCHITECTURES
+        - MORPHEUS_BUILD_PYTHON_STUBS
```
Comment (cwharris) [inline ci/conda/recipes/morpheus/meta.yaml:40]: @mdemoret-nv  Why were these added? They're cmake build args, not env variables.

Comment (cwharris) [reply ci/conda/recipes/morpheus/meta.yaml:40]: I see you've added the expectation of these being env variables as well, however the addition of these two lines is causing a build error.

Comment (mdemoret-nv) [reply ci/conda/recipes/morpheus/meta.yaml:40]: They are not all cmake build args but they do play a part in that. conda-build does not capture the environment from the calling shell by default. If you want to be able to tweak how conda-build is run in any way, you have to specify which environment variables that it should capture and specify them here.

This allows us to tweak how the the morpheus conda package is built from the calling shell. So if you set `CMAKE_CUDA_ARCHITECTURES=NATIVE ./ci/conda/recipes/run_conda_build.sh morpheus` you can speed up the build by only building for your GPU (helpful for testing). Or you can run: `MORPHEUS_BUILD_PYTHON_STUBS=OFF ./ci/conda/recipes/run_conda_build.sh morpheus` to avoid needing the NVIDIA driver during compilation (which we need in the Dockerfile)

Comment (cwharris) [reply ci/conda/recipes/morpheus/meta.yaml:40]: right, but it appears that you now _must_ specify those environment variables in the shell. Is this intentional?
```
Packaging morpheus-0.0.0.dev-cuda_11.5_py38_g2d493e0_71
INFO:conda_build.build:Packaging morpheus-0.0.0.dev-cuda_11.5_py38_g2d493e0_71
Traceback (most recent call last):
  File "/home/charris/miniconda3/bin/conda-mambabuild", line 10, in <module>
    sys.exit(main())
  File "/home/charris/miniconda3/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 239, in main
    call_conda_build(action, config)
  File "/home/charris/miniconda3/lib/python3.9/site-packages/boa/cli/mambabuild.py", line 211, in call_conda_build
    result = api.build(
  File "/home/charris/miniconda3/lib/python3.9/site-packages/conda_build/api.py", line 186, in build
    return build_tree(
  File "/home/charris/miniconda3/lib/python3.9/site-packages/conda_build/build.py", line 3088, in build_tree
    packages_from_this = build(metadata, stats,
  File "/home/charris/miniconda3/lib/python3.9/site-packages/conda_build/build.py", line 2371, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "/home/charris/miniconda3/lib/python3.9/site-packages/conda_build/build.py", line 1602, in bundle_conda
    raise ValueError("env var '{}' specified in script_env, but is not set."
ValueError: env var 'MORPHEUS_BUILD_PYTHON_STUBS' specified in script_env, but is not set.
```

Comment (pdmack) [reply ci/conda/recipes/morpheus/meta.yaml:40]: Works now if I throw 'CMAKE_CUDA_ARCHITECTURES=ALL' in the Dockerfile at the morpheus conda build. 
```
#27 exporting to image
#27 exporting layers
#27 exporting layers 48.8s done
#27 writing image sha256:9773046de3ef35a3c8a5a359d9573b585d51be5f8786d21a181bf7d3529e213e done
#27 naming to nvcr.io/nvidia/morpheus/morpheus:v22.04.00-runtime done
#27 DONE 48.8s
```

**Comment by cwharris** [comment]
Michael's changes LGTM.

**Comment by pdmack** [comment]
Yeah sorry, somehow missed [8a3a6fe](https://github.com/nv-morpheus/Morpheus/pull/164/commits/8a3a6fec86d3dca6d92bd8b449782c088cf8cb82)

**Comment by cwharris** [comment]
rerun tests

**Comment by pdmack** [comment]
LGTM in a local build.

**Review by mdemoret-nv — APPROVED** [review]

**Comment by mdemoret-nv** [comment]
rerun tests

**Comment by mdemoret-nv** [comment]
rerun tests

**Comment by mdemoret-nv** [comment]
rerun tests
